CLN: Remove redundant index test from tests/base/test_ops.py

SaturnFromTitan opened the pull request:
part of #23877
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Boolean indexing is already thoroughly tested in `test_series_mask_boolean` in `tests/indexes/test_na_indexing.py` - [see here](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexing/test_na_indexing.py#L7-L61)

SaturnFromTitan committed:
removed redundant index test from tests/base/test_ops.py

WillAyd reviewed (changes requested):
On board with cleaning up, but are we not losing any coverage here? Maybe overlooking, but one that sticks out is indexing with an Index object which appears here but I think not in the test you linked to replace this?

SaturnFromTitan commented:
Good point @WillAyd, I'll check that later. I was assuming that if we lose coverage, then I'd get a codecov warning here in the PR. Isn't there a bot for these sort of things?

WillAyd commented:
We do have codecov but it’s not generally very insightful (at least for us)

Sent from my iPhone

> On Mar 7, 2020, at 1:24 AM, Martin Winkel <notifications@github.com> wrote:
> 
> ﻿
> Good point @WillAyd, I'll check that later. I was assuming that if we lose coverage, then I'd get a codecov warning here in the PR. Isn't there a bot for these sort of things?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

SaturnFromTitan committed:
Merge branch 'master' into remove-redundant-index-test-from-tests-base

SaturnFromTitan committed:
refactored and split test_series_mask_boolean

SaturnFromTitan committed:
tightened the pytest.raises condition

SaturnFromTitan committed:
renaming and added a comment

SaturnFromTitan committed:
removed some redundancy in the test case

SaturnFromTitan commented on pandas/tests/indexing/test_na_indexing.py:
```diff
@@ -22,42 +25,44 @@
+        msg = "iLocation based boolean indexing cannot use an indexable as a mask"
+        expectation = pytest.raises(ValueError, match=msg)
+    else:
+        expectation = do_not_raise()
```
inspired by https://github.com/pytest-dev/pytest/issues/1830

SaturnFromTitan commented:
@WillAyd I think this is ready for another review now. I added an `indexer_class` parametrization to `test_series_mask_boolean` and refactored the test a little bit (which I mostly had to do to get it passing)

SaturnFromTitan committed:
Revert "removed some redundancy in the test case"

This reverts commit 3b4c786c12a99caec7225df313e85a05b0fd9d1c.

SaturnFromTitan commented on pandas/tests/indexing/test_na_indexing.py:
```diff
@@ -22,42 +23,43 @@
     "mask", [[True, False, False], [True, True, True], [False, False, False]]
 )
-@pytest.mark.parametrize("box_mask", [True, False])
+@pytest.mark.parametrize("indexer_class", [list, pd.array, pd.Index, pd.Series])
```
This test previously only covered `pd.array` and `pd.Series`

SaturnFromTitan commented on pandas/tests/indexing/test_na_indexing.py:
```diff
@@ -7,6 +7,7 @@
 @pytest.mark.parametrize(
     "values, dtype",
     [
+        ([], "object"),
```
The empty scenario was previously covered with some redundant code in the bottom of the test. I moved it to the parametrization instead and adjusted the generic test code to handle it properly

jreback commented:
looks ok, will have to look again after @WillAyd comments are addressed.

SaturnFromTitan commented:
@jreback I addressed all comments of @WillAyd (added the lost test cases to the parametrization of `test_series_mask_boolean`)

jreback commented:
thanks @SaturnFromTitan
